File: simple.cs
class Author {
    public string Name { get; set; }
    public string TwitterHandle { get; set;}
    public string Company { get; set;}

    static public IEnumerable<String> TwitterHandles(IEnumerable<Author> authors, string company) {
        var result = new List<String> ();
        var loopStart = authors; /***/
        foreach (Author a in loopStart) {
            if (a.Company == company) {
                var handle = a.TwitterHandle;
                if (handle != null)
                    result.Add(handle);
            }
        }
        return result;
    }
}

File: simple.py
class Author(object):
    def __init__(self, Name, TwitterHandle, Company):    
        self.Name = Name
        self.TwitterHandle = TwitterHandle
        self.Company = Company

    @staticmethod
    def TwitterHandles(authors, company):
        result = []
        for a in authors:
            if a.Company == company:
                handle = a.TwitterHandle
                if handle != None:
                    result.append(handle)
        return result



a1 = Author('bill', 'bl', 'uperform')
a2 = Author('jacky', 'j', 'uperform')
a3 = Author('papiJam', 'papi', 'logitech')

assert ['bl', 'j'] == Author.TwitterHandles([a1,a2,a3], 'uperform')